src/bit_array.cr

```diff
@@ -54,6 +65,92 @@ struct BitArray
     end
   end
 
+  # Returns all elements that are within the given range.
+  #
+  # Negative indices count backward from the end of the array (-1 is the last
+  # element). Additionally, an empty array is returned when the starting index
+  # for an element range is at the end of the array.
+  #
+  # Raises `IndexError` if the starting index is out of range.
+  #
+  # ```
+  # ba = BitArray.new(5)
+  # ba[0] = true; ba[2] = true; ba[4] = true
+  # ba # => BitArray[10101]
+  #
+  # ba[1..3]    # => BitArray[010]
+  # ba[4..7]    # => BitArray[1]
+  # ba[6..10]   # raise IndexError
+  # ba[5..10]   # => BitArray[]
+  # ba[-2...-1] # => BitArray[0]
+  # ```
+  def [](range : Range(Int, Int))
+    self[*range_to_index_and_count(range)]
+  end
+
+  # Returns count or less (if there aren't enough) elements starting at the
+  # given start index.
+  #
+  # Negative indices count backward from the end of the array (-1 is the last
+  # element). Additionally, an empty array is returned when the starting index
+  # for an element range is at the end of the array.
+  #
+  # Raises `IndexError` if the starting index is out of range.
+  #
+  # ```
+  # ba = BitArray.new(5)
+  # ba[0] = true; ba[2] = true; ba[4] = true
+  # ba # => BitArray[10101]
+  #
+  # ba[-3, 3] # => BitArray[101]
+  # ba[6, 1]  # raise indexError
+  # ba[1, 2]  # => BitArray[01]
+  # ba[5, 1]  # => BitArray[]
+  # ```
+  def [](start : Int, count : Int)
+    raise ArgumentError.new "Negative count: #{count}" if count < 0
+
+    if start == size
+      return BitArray.new(0)
+    end
+
+    start += size if start < 0
+    raise IndexError.new unless 0 <= start <= size
+
+    if count == 0
+      return BitArray.new(0)
+    end
+
+    count = Math.min(count, size - start)
+
+    if @size <= 32
```

Comment:
You always access the attribute `size` through his getter, for consistency you can remove the `@` here